proxy/core/src/main/scala/io/cloudstate/proxy/crdt/CrdtEntity.scala

```diff
@@ -136,6 +135,7 @@ final class CrdtEntity(client: Crdt, configuration: CrdtEntity.Configuration, en
   private[this] final var stopping = false
 
   implicit val ec = context.dispatcher
+  implicit val mat = Materializer(context)
```

Comment:
Who is responsible for managing the lifecycle of instances returned by this method=

Reply:
The actor system. In most cases we're supposed to use the system providing Materializer, do you think this is not one of those cases?

Reply:
It should be possible to infer by the code, for those things taking an implicit materializer—the caller is responsible for the lifecycle management, and for those who create their own materializer, they should be responsible for dealing with it. Just making sure we don't introduce any subtle resource leaks.

Reply:
so ensure shutdown()?

Reply:
Indeed

Reply:
I think we're good. The Materializer(context) within an actor will bind the stream to the actor's lifecycle. The other refactors I did equate to the same thing the deprecated ActorMaterializer() was doing, just refactored to Materializer(system). Either way this creates a new, one-off materializer.

Reply:
@sean-walsh But doesn't this change the semantics? Before the Materializer was a parameter in (hence lifecycle managed on the outside), but now you create a Materializer based on the current ActorContext?

Reply:
This is the shared system materializer. The stream is bound to the actor lifecycle.

Reply:
@sean-walsh No, read the two different apply methods: https://doc.akka.io/api/akka/current/akka/stream/Materializer$.html

Reply:
Ok

Reply:
Blah I mistyped that. The Materializer(context) does create a NEW materializer bound to the actor lifecycle.

Reply:
@sean-walsh Yes, but that is not what the original code does, right? It gets the Materializer passed in from the outside?

Reply:
My thinking was since this is used by the actor, we would use an instance bound to the actor so we don't leave any streams open and this would be best practice.

Reply:
This materializer was also the ActorMaterializer instantiated in CloudStateProxyMain.

Reply:
@sean-walsh So that means it should not be a `Materializer(context)`

Reply:
No, not if you want it to stay as it was, I'll change.

Reply:
One last question, is it ok to use the shared system one or do you want me to use the same one instantiated in CloudStateProxyMain as before and add all the implicit arguments back down the line?

Reply:
@sean-walsh for the Akka 2.6.4 upgrade I'd prefer it to stay as close to what was already there as possible, so we don't introduce subtle bugs that are separate concerns to the 2.6.4 upgrade. We can look at improving/changing the Materializer usage once this upgrade is in.

Reply:
Ok.